### ISSUE TITLE
h5py dependency

### ISSUE THREAD

**Issue by dalcinl** [nschloe/meshio#156]
In commit b5d3cca283a35a84f451b32609841d453964d987, you introduced a hard dependency on h5py, that is, `import meshio` ends up failing with `ImportError` if h5py is not installed. Given that you are listing h5py within `extras_require` in `setup.py`, I guess this hard dependency on h5py was unintentional.

**Comment by nschloe** [comment]
Good catch. I'm not quite sure whether to make it hard dependency now since a couple of mesh formats depend on it. Is it hard to install for you?

**Comment by dalcinl** [comment]
No, installing h5py is not an issue for me, but I hate dependencies. I try to avoid them as much as possible and reduce them at a minimum for my own projects (for example, my mpi4py project does not depend on NumPy, although that would make things much simpler for me and I'm sure no one in the world would object it).

If you want, I can try to put a PR together to make meshio raise `RuntimeError` if h5py is not installed. But you will have to wait a couple of days until I finish my work on #7.

**Comment by nschloe** [comment]
Nah it's alright, I'll do it myself.

The issue with optional dependencies is that pip doesn't install them by default, so after installing meshio, people have to take further steps to make it work for their mesh selection. Almost every mesh i/o module has a dependency now. I've had some complaints about that in the past.

**Comment by dalcinl** [comment]
Maybe you can follow IPython convention and extend `extras_require` with an `all` entry, such that it installs all extra dependencies.
